docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,228 @@
+  },
+  "author": "mkaz",
+  "license": "GPL-2.0-only"
+}
+
+
+Is this OK? (yes) yes
+```
+
+## Webpack & Babel
+
+The next step is to install webpack. You can install packages using the npm command: `npm install`. If you pass the `--save` parameter, npm will write the package as a dependency in package.json file.
+
+Run: `npm install --save webpack`
+
+After installing, a `node_modules` directory is created with the webpack module and its dependencies.
+
+Also, if you look at package.json file it will include a new section:
+```json
+"dependencies": {
+	"webpack": "^4.29.0"
+}
+```
+
+Next, you configure webpack to process the `block.js` file and run babel to transform the JSX.
+
+Create the file `webpack.config.js`
+
+```js
+// sets mode webpack runs under
+const NODE_ENV = process.env.NODE_ENV || 'development';
+
+module.exports = {
+	mode: NODE_ENV,
+
+	// entry is the source script
+	entry: './block.js',
+
+	// output is where to write the built file
+	output: {
+		path: __dirname,
+		filename: 'block.build.js',
+	},
+	module: {
+		// the list of rules used to process files
+		// this looks for .js files, exclude files
+		// in node_modules directory, and uses the
+		// babel-loader to process
+		rules: [
+			{
+				test: /.js$/,
+				exclude: /node_modules/,
+				loader: 'babel-loader',
+			},
+		],
+	},
+};
+```
+
+Next, you need to install babel, the webpack loader, and the JSX plugin using:
+```
+npm install --save babel-loader babel-core babel-plugin-transform-react-jsx
+```
+
+You configure babel by creating a `.babelrc` file:
+```
+{
+	"plugins": [
+		[ "transform-react-jsx", {
+			"pragma": "wp.element.createElement"
+		} ]
+	]
+}
+```
+The pragma setting replaces the function JSX uses for transformation, the `wp.element.createElement` is an equivalent wrapper of `React.createElement` which is the default function for JSX.
+
+With both configs in place, you can now run webpack.
+
+First you need a basic block.js to build. Create `block.js`
+
+```js
+const { registerBlockType } = wp.blocks;
+
+registerBlockType( 'myguten/test-block', {
+	title: 'Basic Example',
+	icon: 'smiley',
+	category: 'layout',
+	edit() {
+		return <div>Hola mundo!</div>;
+	},
+	save() {
+		return <div>Hola mundo!</div>;
+	},
+} );
+```
+
+To configure npm to run a script, you use the scripts section in `package.json` webpack:
+```json
+  "scripts": {
+    "build": "webpack"
+  },
+```
+
+You can then run the build using: `npm run build`
+
+Afterwards, you will see the build file created at: `block.build.js`
+
+## Finishing Touches
+
+### Development Mode
+
+The basics are in place to build. You might of noticed the webpack.config.js set a default mode to "development". Webpack has the ability to run in a "production" mode which shrinks the code down, but makes it difficult to read.
+
+The mode is setup so it can be configured using environment variables, this is easiest to add in the scripts section of `package.json`.
```

Comment:
```suggestion
The mode is setup so it can be configured using environment variables, which can be added in the scripts section of `package.json`.
```